### ISSUE TITLE
Front-end cannot find the file specified jsx

### ISSUE THREAD

**Issue by megadotnet** [lagom/activator-lagom-java-chirper#54]
Front-end cannot find the file specified jsx

## Building environment:

Apache Maven 3.3.9 (bb52d8502b132ec0a5a3f4c09453c07478323dc5; 2015-11-11T00:41:47+08:00)
Maven home: D:\app\apache-maven-3.3.9\bin\..
Java version: 1.8.0_121, vendor: Oracle Corporation
Java home: C:\Program Files\Java\jdk1.8.0_121\jre
Default locale: en_US, platform encoding: GBK
OS name: "windows 7", version: "6.1", arch: "amd64", family: "dos"

## Run following command 

mvn lagom:runAll

Output like this:
[INFO] Copying 1 resource
[INFO] Nothing to compile - all classes are up to date
[INFO] Node v6.10.2 is already installed.
[INFO] Running 'npm install -g react-tools' in H:\Dev\activator-lagom-java-chirper-master\front-end\target\node
[WARNING] npm WARN deprecated react-tools@0.13.3: react-tools is deprecated. For more information, visit https://fb.me/react-tools-deprecated
[INFO] H:\Dev\activator-lagom-java-chirper-master\front-end\target\node\node\jsx -> H:\Dev\activator-lagom-java-chirper-master\front-end\target\node\node\node_modules\react-tools\bin\jsx
[INFO] H:\Dev\activator-lagom-java-chirper-master\front-end\target\node\node
[INFO] `-- react-tools@0.13.3 
[INFO] 
[INFO] Compiling 2 templates...
[INFO] Completed in 230ms
**[ERROR] Command execution failed.
java.io.IOException: Cannot run program "H:\Dev\activator-lagom-java-chirper-master\front-end\target\node\bin\jsx" (in directory "H:\Dev\activator-lagom-java-chirper-master\front-end"): CreateProcess error=2, The system cannot find the file specified
	at java.lang.ProcessBuilder.start(ProcessBuilder.java:1048)
	at java.lang.Runtime.exec(Runtime.java:620)
	at org.apache.commons.exec.launcher.Java13CommandLauncher.exec(Java13CommandLauncher.java:61)**
	at org.apache.commons.exec.DefaultExecutor.launch(DefaultExecutor.java:279)
	at org.apache.commons.exec.DefaultExecutor.executeInternal(DefaultExecutor.java:336)
	at org.apache.commons.exec.DefaultExecutor.execute(DefaultExecutor.java:166)
	at org.codehaus.mojo.exec.ExecMojo.executeCommandLine(ExecMojo.java:804)
	at org.codehaus.mojo.exec.ExecMojo.executeCommandLine(ExecMojo.java:760)
	at org.codehaus.mojo.exec.ExecMojo.execute(ExecMojo.java:304)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
	at com.lightbend.lagom.maven.MavenFacade$$anonfun$executeLifecyclePhase$1$$anonfun$apply$1$$anonfun$apply$mcV$sp$1.apply(MavenFacade.scala:212)
	at com.lightbend.lagom.maven.MavenFacade$$anonfun$executeLifecyclePhase$1$$anonfun$apply$1$$anonfun$apply$mcV$sp$1.apply(MavenFacade.scala:211)
	at scala.collection.Iterator$class.foreach(Iterator.scala:727)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1157)
	at scala.collection.IterableLike$class.foreach(IterableLike.scala:72)
	at scala.collection.AbstractIterable.foreach(Iterable.scala:54)
	at com.lightbend.lagom.maven.MavenFacade$$anonfun$executeLifecyclePhase$1$$anonfun$apply$1.apply$mcV$sp(MavenFacade.scala:211)
	at com.lightbend.lagom.maven.MavenFacade$$anonfun$executeLifecyclePhase$1$$anonfun$apply$1.apply(MavenFacade.scala:205)
	at com.lightbend.lagom.maven.MavenFacade$$anonfun$executeLifecyclePhase$1$$anonfun$apply$1.apply(MavenFacade.scala:205)
	at com.lightbend.lagom.maven.MavenFacade.com$lightbend$lagom$maven$MavenFacade$$switchProject(MavenFacade.scala:228)
	at com.lightbend.lagom.maven.MavenFacade$$anonfun$executeLifecyclePhase$1.apply(MavenFacade.scala:205)
	at com.lightbend.lagom.maven.MavenFacade$$anonfun$executeLifecyclePhase$1.apply(MavenFacade.scala:204)
	at scala.collection.mutable.ResizableArray$class.foreach(ResizableArray.scala:59)
	at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:47)
	at com.lightbend.lagom.maven.MavenFacade.executeLifecyclePhase(MavenFacade.scala:204)
	at com.lightbend.lagom.maven.ServiceManager.com$lightbend$lagom$maven$ServiceManager$$reloadCompile(ServiceManager.scala:169)
	at com.lightbend.lagom.maven.ServiceManager$$anonfun$13.apply(ServiceManager.scala:140)
	at com.lightbend.lagom.maven.ServiceManager$$anonfun$13.apply(ServiceManager.scala:139)
	at com.lightbend.lagom.dev.Reloader$$anonfun$reload$1.apply(Reloader.scala:278)
	at com.lightbend.lagom.dev.Reloader$$anon$4.run(Reloader.scala:44)
	at java.security.AccessController.doPrivileged(Native Method)
	at com.lightbend.lagom.dev.Reloader$.com$lightbend$lagom$dev$Reloader$$withReloaderContextClassLoader(Reloader.scala:40)
	at com.lightbend.lagom.dev.Reloader.reload(Reloader.scala:276)
	at play.core.server.LagomReloadableDevServerStart$$anonfun$mainDev$1$$anon$2$$anonfun$get$1.apply(LagomReloadableDevServerStart.scala:116)
	at play.core.server.LagomReloadableDevServerStart$$anonfun$mainDev$1$$anon$2$$anonfun$get$1.apply(LagomReloadableDevServerStart.scala:114)
	at scala.concurrent.impl.Future$PromiseCompletingRunnable.liftedTree1$1(Future.scala:24)
	at scala.concurrent.impl.Future$PromiseCompletingRunnable.run(Future.scala:24)
	at java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1402)
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)
Caused by: java.io.IOException: CreateProcess error=2, The system cannot find the file specified
	at java.lang.ProcessImpl.create(Native Method)
	at java.lang.ProcessImpl.<init>(ProcessImpl.java:386)
	at java.lang.ProcessImpl.start(ProcessImpl.java:137)
	at java.lang.ProcessBuilder.start(ProcessBuilder.java:1029)
	... 41 more
[INFO] Service friend-impl listening for HTTP on 0:0:0:0:0:0:0:0:60399
[INFO] Service chirp-impl listening for HTTP on 0:0:0:0:0:0:0:0:54485
[INFO] Service activity-stream-impl listening for HTTP on 0:0:0:0:0:0:0:0:51855
[INFO] Service load-test-impl listening for HTTP on 0:0:0:0:0:0:0:0:51796
[INFO] Service front-end listening for HTTP on 0:0:0:0:0:0:0:0:57143
[INFO] (Services started, press enter to stop and go back to the console...)

## Any idea?

**Comment by TimMoore** [comment]
@fsat do you know? I guess the problem is the mismatch between: `H:\Dev\activator-lagom-java-chirper-master\front-end\target\node\node\jsx` and `H:\Dev\activator-lagom-java-chirper-master\front-end\target\node\bin\jsx`

(`node\node` vs. `node\bin`)

I'm not sure how those paths are determined, though.

**Comment by fsat** [comment]
@TimMoore - it's determined by `front-end-maven-plugin` (https://github.com/lagom/activator-lagom-java-chirper/blob/master/front-end/pom.xml#L146 ) when it's installing the `react-tools` (https://github.com/lagom/activator-lagom-java-chirper/blob/master/front-end/pom.xml#L170)

The `jsx` file should be placed in `<front-end basedir>/target/node/bin/jsx` (https://github.com/lagom/activator-lagom-java-chirper/blob/master/front-end/pom.xml#L187). This is where the executable should be placed after calling `npm install`.

I noticed that the failing OS is Windows - I'll try to reproduce this on Windows as well.

**Comment by TimMoore** [comment]
It seems that npm links the executable files to a different location on Windows than other OSes: https://docs.npmjs.com/files/folders#executables

> When in global mode, executables are linked into {prefix}/bin on Unix, or directly into {prefix} on Windows.

Possible workaround would be to specify the full target path to the executable, i.e., `${basedir}/target/node/lib/node_modules/react-tools/bin/jsx`

**Comment by fsat** [comment]
@TimMoore - I've created the PR #55 to fix the issue, but unfortunately I don't have the Windows VM with Maven to test this change. (The only Windows VM I have is for testing/cutting a Python based release, so I'm reluctant to mess around with it).

I was wondering if @megadotnet will be able to quickly checkout the PR #55 branch and then verify if all is well by running `mvn lagom:runAll`.

**Comment by megadotnet** [comment]
@fsat It does not work. Even modified oom.xml file as following:

>                     <configuration>
>                         <executable>${basedir}/target/node/node/node_modules/react-tools/bin/jsx</executable>
>                         <environmentVariables>
>                             <PATH>${env.PATH}:${pom.basedir}/target/node/node</PATH>
>                         </environmentVariables>
>                         <arguments>
>                             <argument>--extension</argument>
>                             <argument>jsx</argument>
>                             <argument>${basedir}/src/main/resources/assets</argument>
>                             <argument>${basedir}/target/classes/public</argument>
>                         </arguments>
>                         
>                         <outputFile>${java.io.tmpdir}/jsx-compile.txt</outputFile>
>                     </configuration>

> mvn clean compile

```
[INFO] Scanning for projects...
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Build Order:
[INFO] 
[INFO] lagom-java-chirper
[INFO] friend-api
[INFO] friend-impl
[INFO] chirp-api
[INFO] chirp-impl
[INFO] activity-stream-api
[INFO] activity-stream-impl
[INFO] load-test-api
[INFO] load-test-impl
[INFO] front-end
[INFO]                                                                         
[INFO] ------------------------------------------------------------------------
[INFO] Building lagom-java-chirper 1.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ lagom-java-chirper ---
[INFO] 
[INFO] --- maven-enforcer-plugin:1.4.1:enforce (enforce-maven-version) @ lagom-java-chirper ---
[INFO]                                                                         
[INFO] ------------------------------------------------------------------------
[INFO] Building friend-api 1.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ friend-api ---
[INFO] Deleting H:\Dev\activator-lagom-java-chirper-master\friend-api\target
[INFO] 
[INFO] --- maven-enforcer-plugin:1.4.1:enforce (enforce-maven-version) @ friend-api ---
[INFO] 
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ friend-api ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory H:\Dev\activator-lagom-java-chirper-master\friend-api\src\main\resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.5.1:compile (default-compile) @ friend-api ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 3 source files to H:\Dev\activator-lagom-java-chirper-master\friend-api\target\classes
[INFO]                                                                         
[INFO] ------------------------------------------------------------------------
[INFO] Building friend-impl 1.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ friend-impl ---
[INFO] Deleting H:\Dev\activator-lagom-java-chirper-master\friend-impl\target
[INFO] 
[INFO] --- maven-enforcer-plugin:1.4.1:enforce (enforce-maven-version) @ friend-impl ---
[INFO] 
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ friend-impl ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 1 resource
[INFO] 
[INFO] --- maven-compiler-plugin:3.5.1:compile (default-compile) @ friend-impl ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 8 source files to H:\Dev\activator-lagom-java-chirper-master\friend-impl\target\classes
[INFO]                                                                         
[INFO] ------------------------------------------------------------------------
[INFO] Building chirp-api 1.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ chirp-api ---
[INFO] Deleting H:\Dev\activator-lagom-java-chirper-master\chirp-api\target
[INFO] 
[INFO] --- maven-enforcer-plugin:1.4.1:enforce (enforce-maven-version) @ chirp-api ---
[INFO] 
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ chirp-api ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory H:\Dev\activator-lagom-java-chirper-master\chirp-api\src\main\resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.5.1:compile (default-compile) @ chirp-api ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 4 source files to H:\Dev\activator-lagom-java-chirper-master\chirp-api\target\classes
[INFO]                                                                         
[INFO] ------------------------------------------------------------------------
[INFO] Building chirp-impl 1.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ chirp-impl ---
[INFO] Deleting H:\Dev\activator-lagom-java-chirper-master\chirp-impl\target
[INFO] 
[INFO] --- maven-enforcer-plugin:1.4.1:enforce (enforce-maven-version) @ chirp-impl ---
[INFO] 
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ chirp-impl ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 1 resource
[INFO] 
[INFO] --- maven-compiler-plugin:3.5.1:compile (default-compile) @ chirp-impl ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 9 source files to H:\Dev\activator-lagom-java-chirper-master\chirp-impl\target\classes
[INFO]                                                                         
[INFO] ------------------------------------------------------------------------
[INFO] Building activity-stream-api 1.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ activity-stream-api ---
[INFO] Deleting H:\Dev\activator-lagom-java-chirper-master\activity-stream-api\target
[INFO] 
[INFO] --- maven-enforcer-plugin:1.4.1:enforce (enforce-maven-version) @ activity-stream-api ---
[INFO] 
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ activity-stream-api ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory H:\Dev\activator-lagom-java-chirper-master\activity-stream-api\src\main\resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.5.1:compile (default-compile) @ activity-stream-api ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 2 source files to H:\Dev\activator-lagom-java-chirper-master\activity-stream-api\target\classes
[INFO]                                                                         
[INFO] ------------------------------------------------------------------------
[INFO] Building activity-stream-impl 1.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ activity-stream-impl ---
[INFO] Deleting H:\Dev\activator-lagom-java-chirper-master\activity-stream-impl\target
[INFO] 
[INFO] --- maven-enforcer-plugin:1.4.1:enforce (enforce-maven-version) @ activity-stream-impl ---
[INFO] 
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ activity-stream-impl ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 1 resource
[INFO] 
[INFO] --- maven-compiler-plugin:3.5.1:compile (default-compile) @ activity-stream-impl ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 2 source files to H:\Dev\activator-lagom-java-chirper-master\activity-stream-impl\target\classes
[INFO]                                                                         
[INFO] ------------------------------------------------------------------------
[INFO] Building load-test-api 1.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ load-test-api ---
[INFO] Deleting H:\Dev\activator-lagom-java-chirper-master\load-test-api\target
[INFO] 
[INFO] --- maven-enforcer-plugin:1.4.1:enforce (enforce-maven-version) @ load-test-api ---
[INFO] 
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ load-test-api ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory H:\Dev\activator-lagom-java-chirper-master\load-test-api\src\main\resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.5.1:compile (default-compile) @ load-test-api ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 2 source files to H:\Dev\activator-lagom-java-chirper-master\load-test-api\target\classes
[INFO]                                                                         
[INFO] ------------------------------------------------------------------------
[INFO] Building load-test-impl 1.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ load-test-impl ---
[INFO] Deleting H:\Dev\activator-lagom-java-chirper-master\load-test-impl\target
[INFO] 
[INFO] --- maven-enforcer-plugin:1.4.1:enforce (enforce-maven-version) @ load-test-impl ---
[INFO] 
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ load-test-impl ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 1 resource
[INFO] 
[INFO] --- maven-compiler-plugin:3.5.1:compile (default-compile) @ load-test-impl ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 3 source files to H:\Dev\activator-lagom-java-chirper-master\load-test-impl\target\classes
[INFO]                                                                         
[INFO] ------------------------------------------------------------------------
[INFO] Building front-end 1.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ front-end ---
[INFO] Deleting H:\Dev\activator-lagom-java-chirper-master\front-end\target
[INFO] 
[INFO] --- maven-enforcer-plugin:1.4.1:enforce (enforce-maven-version) @ front-end ---
[INFO] 
[INFO] --- frontend-maven-plugin:1.4:install-node-and-npm (install node and npm) @ front-end ---
[INFO] Installing node version v6.10.2
[INFO] Unpacking C:\Users\Megadotnet\.m2\repository\com\github\eirslett\node\6.10.2\node-6.10.2-win-x64.zip into H:\Dev\activator-lagom-java-chirper-master\front-end\target\node\node\tmp
[INFO] Copying node binary from H:\Dev\activator-lagom-java-chirper-master\front-end\target\node\node\tmp\node-v6.10.2-win-x64\node.exe to H:\Dev\activator-lagom-java-chirper-master\front-end\target\node\node\node.exe
[INFO] Installed node locally.
[INFO] 
[INFO] --- frontend-maven-plugin:1.4:npm (install .jsx compiler) @ front-end ---
[INFO] Running 'npm install -g react-tools' in H:\Dev\activator-lagom-java-chirper-master\front-end\target\node
[WARNING] npm WARN deprecated react-tools@0.13.3: react-tools is deprecated. For more information, visit https://fb.me/react-tools-deprecated
[INFO] H:\Dev\activator-lagom-java-chirper-master\front-end\target\node\node\jsx -> H:\Dev\activator-lagom-java-chirper-master\front-end\target\node\node\node_modules\react-tools\bin\jsx
[INFO] H:\Dev\activator-lagom-java-chirper-master\front-end\target\node\node
[INFO] `-- react-tools@0.13.3 
[INFO]   +-- commoner@0.10.8 
[INFO]   | +-- commander@2.9.0 
[INFO]   | | `-- graceful-readlink@1.0.1 
[INFO]   | +-- detective@4.5.0 
[INFO]   | | +-- acorn@4.0.11 
[INFO]   | | `-- defined@1.0.0 
[INFO]   | +-- glob@5.0.15 
[INFO]   | | +-- inflight@1.0.6 
[INFO]   | | | `-- wrappy@1.0.2 
[INFO]   | | +-- inherits@2.0.3 
[INFO]   | | +-- minimatch@3.0.4 
[INFO]   | | | `-- brace-expansion@1.1.7 
[INFO]   | | |   +-- balanced-match@0.4.2 
[INFO]   | | |   `-- concat-map@0.0.1 
[INFO]   | | +-- once@1.4.0 
[INFO]   | | `-- path-is-absolute@1.0.1 
[INFO]   | +-- graceful-fs@4.1.11 
[INFO]   | +-- iconv-lite@0.4.17 
[INFO]   | +-- mkdirp@0.5.1 
[INFO]   | | `-- minimist@0.0.8 
[INFO]   | +-- private@0.1.7 
[INFO]   | +-- q@1.5.0 
[INFO]   | `-- recast@0.11.23 
[INFO]   |   +-- ast-types@0.9.6 
[INFO]   |   +-- esprima@3.1.3 
[INFO]   |   `-- source-map@0.5.6 
[INFO]   `-- jstransform@10.1.0 
[INFO]     +-- base62@0.1.1 
[INFO]     +-- esprima-fb@13001.1001.0-dev-harmony-fb 
[INFO]     `-- source-map@0.1.31 
[INFO]       `-- amdefine@1.0.1 
[INFO] 
[INFO] 
[INFO] --- twirl-maven-plugin:1.0.5:compile (default) @ front-end ---
[INFO] Compiling 2 templates...
[INFO] Completed in 2683ms
[INFO] 
[INFO] --- exec-maven-plugin:1.6.0:exec (compile .jsx files) @ front-end ---
[ERROR] Command execution failed.
java.io.IOException: Cannot run program "H:\Dev\activator-lagom-java-chirper-master\front-end\target\node\node\node_modules\react-tools\bin\jsx" (in directory "H:\Dev\activator-lagom-java-chirper-master\front-end"): CreateProcess error=193, %1 is not a valid Win32 application
	at java.lang.ProcessBuilder.start(ProcessBuilder.java:1048)
	at java.lang.Runtime.exec(Runtime.java:620)
	at org.apache.commons.exec.launcher.Java13CommandLauncher.exec(Java13CommandLauncher.java:61)
	at org.apache.commons.exec.DefaultExecutor.launch(DefaultExecutor.java:279)
	at org.apache.commons.exec.DefaultExecutor.executeInternal(DefaultExecutor.java:336)
	at org.apache.commons.exec.DefaultExecutor.execute(DefaultExecutor.java:166)
	at org.codehaus.mojo.exec.ExecMojo.executeCommandLine(ExecMojo.java:804)
	at org.codehaus.mojo.exec.ExecMojo.executeCommandLine(ExecMojo.java:760)
	at org.codehaus.mojo.exec.ExecMojo.execute(ExecMojo.java:304)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:207)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:80)
	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:307)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:193)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:106)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:863)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:288)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:199)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
Caused by: java.io.IOException: CreateProcess error=193, %1 is not a valid Win32 application
	at java.lang.ProcessImpl.create(Native Method)
	at java.lang.ProcessImpl.<init>(ProcessImpl.java:386)
	at java.lang.ProcessImpl.start(ProcessImpl.java:137)
	at java.lang.ProcessBuilder.start(ProcessBuilder.java:1029)
	... 30 more
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
[INFO] lagom-java-chirper ................................. SUCCESS [  0.747 s]
[INFO] friend-api ......................................... SUCCESS [  2.452 s]
[INFO] friend-impl ........................................ SUCCESS [  1.225 s]
[INFO] chirp-api .......................................... SUCCESS [  0.450 s]
[INFO] chirp-impl ......................................... SUCCESS [  0.630 s]
[INFO] activity-stream-api ................................ SUCCESS [  0.516 s]
[INFO] activity-stream-impl ............................... SUCCESS [  0.392 s]
[INFO] load-test-api ...................................... SUCCESS [  0.215 s]
[INFO] load-test-impl ..................................... SUCCESS [  0.522 s]
[INFO] front-end .......................................... FAILURE [01:22 min]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 01:30 min
[INFO] Finished at: 2017-05-09T09:37:23+08:00
[INFO] Final Memory: 73M/447M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.codehaus.mojo:exec-maven-plugin:1.6.0:exec (compile .jsx files) on project front-end: Command execution failed. Cannot run program "H:\Dev\activator-lagom-java-chirper-master\front-end\target\node\node\node_modules\react-tools\bin\jsx" (in directory "H:\Dev\activator-lagom-java-chirper-master\front-end"): CreateProcess error=193, %1 is not a valid Win32 application -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <goals> -rf :front-end
```

**Comment by fsat** [comment]
@megadotnet - Thanks for trying it out - I really appreciate it. I think this is caused by `jsx` executable not being able to be run on Windows:

```
Caused by: java.io.IOException: CreateProcess error=193, %1 is not a valid Win32 application
	at java.lang.ProcessImpl.create(Native Method)
	at java.lang.ProcessImpl.<init>(ProcessImpl.java:386)
	at java.lang.ProcessImpl.start(ProcessImpl.java:137)
	at java.lang.ProcessBuilder.start(ProcessBuilder.java:1029)
```
I'll see if there's a jsx compiler for windows, and if all else fails I'll need to see if the latest `.jsx` compiler from Babel works on Windows:

https://babeljs.io/docs/plugins/preset-react/

I'm somewhat reluctant to use the Babel since it pulls down significantly more dependencies, and thus slowing down the `mvn compile` and the hot reload with the way the frontend plugin is used.

cc @TimMoore

**Comment by TimMoore** [comment]
Does it need to invoke `jsx.bat` on Windows?

I'm not 100% across how things work on Windows, but I wouldn't be surprised if you need to include the extension when calling by a full path name.

I'm just guessing, though. We'll need someone running Windows to work out a solution I think.

**Comment by benmccann** [comment]
@megadotnet could you try out https://github.com/lagom/activator-lagom-java-chirper/pull/58 ?

**Comment by TimMoore** [comment]
Fixed by #58

**Comment by megadotnet** [comment]
@benmccann  It was fixed.
